Automattic/jetpack
# Backup package: Refactor Admin and ReviewRestore components

#### Changes proposed in this Pull Request:
Refactor ReviewRestore component by making it a Presentational component. 
Refactor Admin component by removing internal functions.

Additionally, fix two issues:
1- issue pointed out by @oskosk in [this comment](https://github.com/Automattic/jetpack/pull/25280#issuecomment-1197093437), where dismissing the review request was returning this error:
![image](https://user-images.githubusercontent.com/16329583/181854524-a5f5df89-d704-4f6c-ab20-06c3e1d64f81.png)
Addressed by adding a return to `manage_dismissed_backup_review_request`

2- Clicking on the dismiss button was also taking to the top of the page, now addressed by `e.preventDefault();`



#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:






* Create a Jurassic Ninja site, add a Backup plan and add your site credentials to Jetpack Backup
* Run a restore for your site
* Confirm that the message asking for a request shows on your Jetpack Backup Admin page
* Click on the link and confirm that you're taken to the WordPress.org Jetpack Backup plugin review page
* Click on dismiss and confirm that the review request disappears
* Reload the page and confirm that the message is not there anymore
